### PR TITLE
Decode an open tool call past a stop token

### PR DESCRIPTION
**Pull request by alexph-dev**
A model that samples a stop token inside an open `<tool_call>` fails the request.

Seen with Qwen3.8-Flash-Next 125B-A6B Q2: after 351 generated tokens the model emitted `<|im_end|>` while the block was still open.

    ds4-server: incomplete DeepSeek tool call: stop=stop token token=248046 generated=351 limit=32768 room=220010
    ds4-server: chat ... DSML_START finish=error error="unterminated tool call (stop token; token=248046, generated=351, limit=32768)"

`try_repair_dsml()` only restores a missing outer wrapper, and the tool-error continuation is skipped for streaming clients, so a streaming client loses the turn and has to retry the whole request.

A tool block carries no executable call until it closes, so a stop token there is invalid output rather than the end of the turn. Decoding already constrains that region (greedy outside payload sampling); this keeps decoding with the best non-stop token instead of accepting the stop, using the existing `ds4_session_argmax_ignoring_eos()`. Drafted stop tokens drop through the existing rewind path so the next sample sees fresh logits.

The override is bounded by `DS4_TOOL_CALL_STOP_SUPPRESS_LIMIT` (24) so a model that insists on ending inside the block still terminates into the existing error path instead of spending the whole output budget. When a block never closes, the warning now carries the tail of the generation, which is what was missing while diagnosing this.

Testing: macOS 26.5.2 on an M5 Max (128 GB), Metal backend, Qwen3.8-Flash-Next 125B-A6B Q2.

    make ds4_test && ./ds4_test --server   # ok
    make ds4-server                        # clean, no new warnings

Live checks against the running server: a Responses tool call, a Responses tool-output continuation, an OpenAI chat stream and an Anthropic stream all behave as before, and a Codex session completes a full reasoning -> tool call -> answer turn.

One of three independent `ds4_server.c` fixes from the same debugging session, alongside #15 and #16. They do not depend on each other and can be taken in any order.

### CODE REVIEW

**Comment by alexph-dev** [comment]
Closing: reopened against the upstream project as antirez/ds4#1071, since this repository tracks antirez/ds4 main exactly.
